test/decorator-observable.spec.js

```diff
@@ -108,15 +111,69 @@ describe('observable decorator', () => {
     instance.value = newValue;
     expect(instance.customHandler).toHaveBeenCalledWith(newValue, oldValue, 'value');
   });
-  
+
   it('should create an enumerable accessor', () => {
     const instance = new class {
       @observable value;
     };
-    const keys = [];
-    for (let p in instance) {
-      keys.push(p);
-    }
-    expect(keys).toContain('value');
+
+    expect(Object.getPrototypeOf(instance).propertyIsEnumerable('value')).toBe(true);
+  });
+
+  describe('private property', () => {
+    it('should exist', () => {
+      const instance = new class {
+        @observable value;
+      };
+
+      expect(Object.getPrototypeOf(instance).hasOwnProperty('_value')).toBe(true);
+    });
+
+    it('should be non-enumerable', () => {
+      const instance = new class {
+        @observable value;
+      };
+
+      expect(Object.getPrototypeOf(instance).propertyIsEnumerable('_value')).toBe(false);
+    });
+
+    describe('observation accessors', () => {
+      const instance = new class {
+        @observable value;
+      };
+
+      const observer = new SetterObserver(
+        new TaskQueue(),
+        Object.getPrototypeOf(instance),
+        '_value'
+      );
+
+      it('should convert', () => {
+        spyOn(Logger.prototype, 'warn');
+        observer.convertProperty();
+        expect(Logger.prototype.warn).not.toHaveBeenCalled();
+      });
+
+      it('should exist', () => {
```

Comment:
This test asserts that `SetterObserver` creates the getter-setter on the right object (https://github.com/aurelia/binding/pull/506#issuecomment-250547363).